lib/page-object/sections.rb

```diff
@@ -0,0 +1,29 @@
+module PageObject
+  class SectionCollection
+    include Enumerable
+
+    def initialize sections
+      @sections = sections
+    end
+
+    def each &block
+      @sections.each &block
+    end
+
+    def [] index
+      @sections[index]
+    end
+
+    def find_by values_hash
```

Comment:
find_by and select_by are essentially find and select where you provide key value pairs that correspond to methods and return values. The comparison is === so it has flexibilities for regex and such.